torch/distributed/rpc/backend_registry.py

```diff
@@ -21,14 +22,56 @@ def register_backend(backend_name, init_backend_handler):
             `_init_rpc()` function is called with a backend.
              This returns the agent.
     """
-    backend_registry = _get_backend_registry()
-    if backend_name in backend_registry:
+    global BackendType
+    if backend_name in BackendType.__members__.keys():
         raise RuntimeError("RPC backend {}: already registered".format(backend_name))
-    backend_registry[backend_name] = init_backend_handler
+    # Create a new enum type, `BackendType`, with extended members.
+    existing_enum_dict = {member.name: member.value for member in BackendType}
+    extended_enum_dict = dict(
+        {backend_name: BackendValue(init_backend_handler=init_backend_handler)},
+        **existing_enum_dict
+    )
+    BackendType = enum.Enum(value="BackendType", names=extended_enum_dict)
+    return BackendType[backend_name]
+
+
+def init_backend(backend, *args, **kwargs):
+    return backend.value.init_backend_handler(*args, **kwargs)
+
+
+def process_group_init_backend_handler(
+    store,
+    self_name,
+    self_rank,
+    worker_name_to_id,
+    num_send_recv_threads,
+    *args,
+    **kwargs
+):
+    # Initialize ProcessGroup.
+    if not dc10d.is_initialized():
+        world_size = len(worker_name_to_id)
+        dc10d.init_process_group(
+            backend="gloo", store=store, rank=self_rank, world_size=world_size
+        )
+
+    group = dc10d._get_default_group()
```

Comment:
Is this the only function that's not available in `torch.distributed` and used in this file? If yes, shall we import this function explicitly from `distributed_c10d`. The other two functions `is_initialized` and `init_process_group` can be called from `torch.distributed`.

Reply:
Updated. Only this function is called through `distributed_c10d` module.